vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/ad_targeting.cc

```diff
@@ -48,4 +48,32 @@ SegmentList AdTargeting::GetSegments() const {
   return segments;
 }
 
+SegmentList AdTargeting::GetInterestSegments() const {
+  SegmentList segments;
+
+  if (features::IsTextClassificationEnabled()) {
+    const ad_targeting::model::TextClassification text_classification_model;
+    const SegmentList text_classification_segments =
+        text_classification_model.GetSegments();
```

Comment:
Maybe we can just return `text_classification_model.GetSegments();` instead of copying to segments